README.md

```diff
@@ -1,5 +1,119 @@
 # gen-typescript-declarations
 
 [![Build Status](https://travis-ci.org/Polymer/gen-typescript-declarations.svg?branch=master)](https://travis-ci.org/Polymer/gen-typescript-declarations)
+[![NPM version](https://img.shields.io/npm/v/@polymer/gen-typescript-declarations.svg)](https://www.npmjs.com/package/@polymer/gen-typescript-declarations)
 
-Generate TypeScript declarations from Polymer Analyzer output
+A library which generates TypeScript declarations for Polymer and custom
+elements.
+
+## How do I use the typings?
+
+### Polymer
+
+Typings for the Polymer 2.0 library will be included in future releases on
+Bower in the `types/` directory ([pending
+PR](https://github.com/Polymer/polymer/pull/4928)).
+
+Install Polymer normally from Bower, and add a [triple-slash
+directive](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html)
+anywhere in your TypeScript project for the typings you require. Each HTML
+import from Polymer has a corresponding typings file. For example, if you
+depend on `polymer-element.html`:
+
+```ts
+/// <reference path="./bower_components/polymer/types/polymer-element.d.ts" />`
+
+class MyElement extends Polymer.Element {
+  ...
+}
+```
+
+You may also be interested in the [Polymer
+decorators](https://github.com/Polymer/polymer-decorators).
+
+## How do I generate new typings?
+
+You can run this package from the command line with `gen-tsd`, or as a library
+with the `generateDeclarations` function.
+
+It is recommended to integrate typings generation as part of your build/release
+process:
+
+```sh
+$ npm install --save-dev @polymer/gen-typescript-declarations
+```
+
+Add a `gen-tsd` script to your `package.json`:
+
+```js
+{
+  ...
+  "scripts": {
+    "gen-tsd": "gen-tsd"
+  }
+}
+```
+
+If you're using Bower, ensure you run `npm run gen-tsd` to generate the latest
+typings and commit them to your repository before tagging each release.
+
+If you're using NPM, you can instead add this script to the NPM `prepack`
+script to generate and include typings in your NPM package every time you
+publish.
+
+## Config options
+
+By default the `gen-tsd` command will read a file called `gen-tsd.json` in
+your root directory. It has the following options:
+
+* **`exclude`**`: string[]`
+
+  Skip source files whose paths match this pattern. If `undefined`, defaults
```

Comment:
What kind of pattern? Minimatch?

Reply:
Since https://github.com/Polymer/gen-typescript-declarations/pull/44 it is minimatch. Updated description.